src/controllers/shared.controllers/director.correspondence.address.controller.ts

```diff
@@ -91,33 +94,48 @@ export const postDirectorCorrespondenceAddress = async (req: Request, res: Respo
     };
 
     const canUseRegisteredOfficeAddress = verifyUseRegisteredOfficeAddress(selectedSraAddressChoice, companyProfile, officerFilingBody);
-    
-    const patchFiling = await patchOfficerFiling(session, transactionId, submissionId, officerFilingBody);
-    let path: string;
-
-    if (!canUseRegisteredOfficeAddress && selectedSraAddressChoice === registeredOfficerAddressValue) {
-      path = isUpdate ? UPDATE_DIRECTOR_LINK_CORRESPONDENCE_ADDRESS_ENTER_MANUALLY_PATH : DIRECTOR_LINK_CORRESPONDENCE_ADDRESS_ENTER_MANUALLY_PATH;
-      return redirectToPath(path, req, res, lang);
-    }
+    if (canUseRegisteredOfficeAddress && isUpdate) {
 
-    if (patchFiling.data.isHomeAddressSameAsServiceAddress && selectedSraAddressChoice === registeredOfficerAddressValue) {
-      path = isUpdate ? UPDATE_DIRECTOR_RESIDENTIAL_ADDRESS_PATH : DIRECTOR_RESIDENTIAL_ADDRESS_PATH;
-      return redirectToPath(path, req, res, lang);
-    }
+      const officerFiling = await getOfficerFiling(session, transactionId, submissionId);
+      const appointmentId = officerFiling.referenceAppointmentId as string;
+      const companyAppointment: CompanyAppointment = await getCompanyAppointmentFullRecord(session, companyNumber, appointmentId);
 
-    if (!patchFiling.data.isHomeAddressSameAsServiceAddress && selectedSraAddressChoice === registeredOfficerAddressValue) {
-      path = isUpdate ? UPDATE_DIRECTOR_CORRESPONDENCE_ADDRESS_LINK_PATH : DIRECTOR_CORRESPONDENCE_ADDRESS_LINK_PATH;
-      return redirectToPath(path, req, res, lang);
+      officerFilingBody.correspondenceAddressHasBeenUpdated = checkIsCorrespondenceAddressUpdated(
```

Comment:
Is the filing updated during this post? I don't believe a valid address is entered at this stage so we wouldn't want to mark this as a pending update.

Reply:
yes, that's the bit I missed previously. When complete ROA we save the residential address into the filing. Which can be viewed as an address change - even if the next page decides if the roa we put in service_address is actually used.